public/app/features/alerting/unified/utils/matchers.ts

```diff
@@ -24,3 +26,36 @@ export const getMatcherQueryParams = (labels: Labels) => {
 
   return matcherUrlParams;
 };
+
+export const findAlertRulesWithMatchers = (rules: CombinedRule[], matchers: MatcherFieldValue[]) => {
+  const hasMatcher = (rule: CombinedRule, matcher: MatcherFieldValue) => {
+    return Object.entries(rule.labels).some(([key, value]) => {
```

Comment:
There is a very similar piece of code in the `labelsMatchMatchers` function from `public/app/features/alerting/unified/utils/alertmanager.ts`. Maybe it's worth reusing it?

Reply:
Nice! Although it did not fit this use case.